presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -721,16 +746,24 @@ private void revokeRole(String role, String granteeName, PrincipalType granteeTy
         }
     }
 
-    private List<String> getPrestoViews(String databaseName)
+    private List<String> doGetTablesWithParameter(String databaseName, String parameterKey, String parameterValue)
             throws TException
     {
+        /*
+         * The parameter value is restricted to have only alphanumeric characters so that it's safe to be used against HMS.
+         * When using with a LIKE operator, the HMS may want the parameter value to follow a Java regex pattern or a SQL
+         * pattern. And it's hard to predict the HMS's behavior from outside.
+         */
+        if (!TABLE_PARAMETER_VALUE_PATTERN.matcher(parameterValue).matches()) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, "The parameter value has one or more non-alphanumeric characters");
```

Comment:
`format("The parameter value '%s' does not match the pattern '%s'",  value, TABLE_PARAMETER_SAFE_VALUE_PATTERN.pattern())`

Also, i'd use `checkArgument` here, since the error being thrown is not supposed to reach end-user. It may be raised only when there is a bug in Presto.